platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationContextImpl.java

```diff
@@ -196,6 +196,47 @@ public boolean cleanDirectory(Path path) {
     return true;
   }
 
+  /**
+   * Temporary version of cleanDirectory() that doesn't delete any directories but only files found.
+   * This is to work around a directory handle issue on Windows which leaves the directory in a
+   * weird state and prevents us from recreating it.
+   *
+   * <p>Cleans a directory without deleting it.
+   *
+   * @param directory directory to clean
+   * @throws IOException in case cleaning is unsuccessful
+   */
+  private static void cleanDirectory(final File directory) throws IOException {
+    final File[] files = directory.listFiles();
+
+    if (files == null) { // null if security restricted
+      throw new IOException("failed to list contents of " + directory);
+    }
+    IOException ioe = null;
+
+    for (final File file : files) {
+      try {
+        if (file.isDirectory()) {
+          ImportMigrationContextImpl.cleanDirectory(file);
+        } else {
+          final boolean exists = file.exists();
+
+          if (!file.delete()) {
+            if (!exists) {
+              throw new FileNotFoundException("file does not exist: " + file);
+            }
+            throw new IOException("unable to delete file: " + file);
```

Comment:
❓ Should we log something when these happen?

Reply:
Handled by the calling code. This is meant to replace Apache's cleanDirectory() method so I do not want t log in here and start complicating this method since it will go away with the next PR for this ticket.